### PR TITLE
Release v2.4.0

### PR DESCRIPTION
**Pull request by alexmojaki**
Bumping version to v2.4.0.

### CODE REVIEW

File: logfire-api/logfire_api/_internal/main.pyi [hunk 1]
```diff
@@ -223,28 +223,10 @@ class Logfire:
...
-            span_name: The span name. If not provided, the `msg_template` will be used.
-            extract_args: By default, all function call arguments are logged as span attributes.
-                Set to `False` to disable this, or pass an iterable of argument names to include.
-            allow_generator: Set to `True` to prevent a warning when instrumenting a generator function.
-                Read https://logfire.pydantic.dev/docs/guides/advanced/generators/#using-logfireinstrument first.
-        """
+    @overload
+    def instrument(self, msg_template: LiteralString | None = None, *, span_name: str | None = None, extract_args: bool | Iterable[str] = True, allow_generator: bool = False) -> Callable[[Callable[P, R]], Callable[P, R]]: ...
+    @overload
+    def instrument(self, func: Callable[P, R]) -> Callable[P, R]: ...
```
Comment (alexmojaki) [inline logfire-api/logfire_api/_internal/main.pyi]: @Kludex apparently stubgen doesn't work with docstrings and overloads?

Comment (Kludex) [reply logfire-api/logfire_api/_internal/main.pyi]: We need to add the flag `--include-docstrings`.

Comment (Kludex) [reply logfire-api/logfire_api/_internal/main.pyi]: Ah... Sorry. Got it.

Comment (Kludex) [reply logfire-api/logfire_api/_internal/main.pyi]: Sad.

Comment (Kludex) [reply logfire-api/logfire_api/_internal/main.pyi]: Maybe because of the wrong `func` name?

Comment (Kludex) [reply logfire-api/logfire_api/_internal/main.pyi]: What if the `func` becomes `msg_template`, does it work?

Comment (alexmojaki) [reply logfire-api/logfire_api/_internal/main.pyi]: no that doesn't help.

```python
def overload(func):
    """Decorator for overloaded functions/methods.

    In a stub file, place two or more stub definitions for the same
    function in a row, each decorated with @overload.

    For example::

        @overload
        def utf8(value: None) -> None: ...
        @overload
        def utf8(value: bytes) -> bytes: ...
        @overload
        def utf8(value: str) -> bytes: ...

    In a non-stub file (i.e. a regular .py file), do the same but
    follow it with an implementation.  The implementation should *not*
    be decorated with @overload::

        @overload
        def utf8(value: None) -> None: ...
        @overload
        def utf8(value: bytes) -> bytes: ...
        @overload
        def utf8(value: str) -> bytes: ...
        def utf8(value):
            ...  # implementation goes here
```

so stubgen is doing what it's supposed to do.

but ruff doesn't want me to put docstrings on overloads: https://docs.astral.sh/ruff/rules/overload-with-docstring/

so i don't know what the expectation here is. i guess we leave it?

Comment (Kludex) [reply logfire-api/logfire_api/_internal/main.pyi]: drop that ruff rule

**Review by Kludex — APPROVED** [review]

**Pull request merged**